doc/languages-frameworks/go.section.md

```diff
@@ -13,6 +13,7 @@ In the following is an example expression using `buildGoModule`, the following a
 
 - `vendorSha256`: is the hash of the output of the intermediate fetcher derivation. `vendorSha256` can also take `null` as an input. When `null` is used as a value, rather than fetching the dependencies and vendoring them, we use the vendoring included within the source repo. If you'd like to not have to update this field on dependency changes, run `go mod vendor` in your source repo and set `vendorSha256 = null;`
 - `runVend`: runs the vend command to generate the vendor directory. This is useful if your code depends on c code and go mod tidy does not include the needed sources to build.
+- `proxyVendor`: Fetches (go mod download) and proxies the vendor directory. This is useful if any dependency has case-insensitive conflicts.
```

Comment:
I wonder if we want to transition to this option eventually.

Reply:
`go mod download` / `GOPROXY` is what `buildGoModule` used previously, it really isn't an improvement except that it does handle stuff like this. If we want to simplify this builder I suppose we could remove `runVend` as `proxyVendor` will also cover that use case.

Reply:
But still if we would have it by default would it not remove a footgun? We often do not run builds on macOS hence it would be hard to detect this issue.

Reply:
Yes it would but it is a non-trivial performance cost as the modules are downloaded twice. This has been a problem since we switched to `go mod vendor` and we've only had two occurrences (that we know of) so I don't think it's worth it.

Reply:
This happens a lot more often than you think but it is usually ignored because most people don't care enough about darwin.

Reply:
Also I think we should add that you should try to set this when you get a vendor hash mismatch on darwin.

Reply:
> This happens a lot more often than you think

How often?

We currently have ~600 packages that use the two phase build. It would need to be a relatively common problem to justify a permanent ~2x performance hit during the first phase on all of them. If it is that common we may be better off trying to find another way of dealing with this.

Reply:
> How often?

I think I encountered it two times the last week. Can't remember which package but we shouldn't enable it by default.

Reply:
> Also I think we should add that you should try to set this when you get a vendor hash mismatch on darwin.

Done.

Reply:
> If we want to simplify this builder I suppose we could remove `runVend` as `proxyVendor` will also cover that use case.

I'll leave this for another PR, I want to have `proxyVendor` in master for a bit before making further changes.

Reply:
Follow up PR: https://github.com/NixOS/nixpkgs/pull/152935